apis/v1beta1/gateway_types.go

```diff
@@ -476,6 +476,14 @@ type AllowedRoutes struct {
 	//
 	// +optional
 	// +kubebuilder:validation:MaxItems=8
+	//
+	// The following list states the Route kinds compatible for each protocolType
+	//
+	// HTTP: HTTPRoute, GRPCRoute
+	// HTTPS: HTTPRoutes, GRPCRoutes
+	// TCP: TCPRoute
+	// UDP:	UDPRoute
+	// TLS: TLSRoute
```

Comment:
TLS protocol + TCPRoute should also be supported, right? (ref. https://gateway-api.sigs.k8s.io/guides/tls/ and https://gateway-api.sigs.k8s.io/concepts/api-overview/#route-summary-table)